composeApp/src/commonMain/kotlin/ui/components/dialogs/ScaleFontSizeDialog.kt

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2024 The ZZZ Archive Open Source Project by mrfatworm
+ * License: MIT License
+ */
+
+package ui.components.dialogs
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.widthIn
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import org.jetbrains.compose.resources.painterResource
+import org.jetbrains.compose.resources.stringResource
+import ui.components.ZzzSlider
+import ui.components.buttons.ZzzOutlineButton
+import ui.components.buttons.ZzzPrimaryButton
+import ui.components.items.RarityItem
+import ui.theme.AppTheme
+import zzzarchive.composeapp.generated.resources.Res
+import zzzarchive.composeapp.generated.resources.apply
+import zzzarchive.composeapp.generated.resources.default_value
+import zzzarchive.composeapp.generated.resources.font_scale
+import zzzarchive.composeapp.generated.resources.img_summer_solstice_2023_artwork
+import zzzarchive.composeapp.generated.resources.multiply
+import zzzarchive.composeapp.generated.resources.ui_scale
+import zzzarchive.composeapp.generated.resources.ui_scale_warning
+import kotlin.math.round
+
+@Composable
+fun ScaleFontSizeDialog(
+    uiScaleValue: Float,
+    fontScaleValue: Float,
+    onApply: (Float, Float) -> Unit,
+    onDismiss: () -> Unit
+) {
+    var uiScale by AppTheme.uiScale
+    var fontScale by AppTheme.fontScale
+    Dialog(onDismissRequest = {
+        uiScale = uiScaleValue
+        fontScale = fontScaleValue
+        onDismiss()
+    }) {
+        Card(
```

Comment:
[Nit] Consider breaking down this components into smaller ones. I think it could be handle a lot of responsibilities.